src/java.prefs/macosx/native/libprefs/MacOSXPreferencesFile.m

```diff
@@ -695,9 +695,9 @@ static void createTreeForPath(CFStringRef path, CFStringRef name,
     beforeAdd = CFDictionaryContainsKey(parent, child);
     CFDictionaryAddValue(parent, child, node);
     if (!beforeAdd)
-        beforeAdd = CFDictionaryContainsKey(parent, child);
+        beforeAdd = CFDictionaryContainsKey(parent, child) ? JNI_TRUE : JNI_FALSE;
```

Comment:
If you do this here you need something similar on line 695. Still say it is simpler to use Boolean internally and convert to jboolean on return expression.